src/modules/socket.js

```diff
@@ -206,6 +206,35 @@ const emitChatEvent = async (io, chat) => {
   }
 };
 
+const emitUpdateEvent = async (io, roomId) => {
+  try {
+    // 방의 모든 사용자에게 socket 메세지 업데이트 이벤트를 발생시킵니다.
+    if (!io || !roomId) {
+      throw new IllegalArgumentsException();
+    }
+
+    const { name, part } = await roomModel.findById(roomId, "name part");
+
+    if (!name || !part) {
+      throw new IllegalArgumentsException();
+    }
+
+    const userIds = part.map((participant) => participant.user);
+    await Promise.all(
+      userIds.map(async (userId) =>
+        io.in(`user-${userId}`).emit("chat_update", {
+          roomId,
+        })
+      )
+    );
```

Comment:
`async` 함수로 만들지 않아도 될 것 같네요 ㅎㅎㅎ

```suggestion
    part.forEach(({ user }) =>
      io.in(`user-${user}`).emit("chat_update", { roomId })
    )
```

Reply:
방 인원이 어차피 많아봤자 4명이라 비동기안해도 괜찮다는 것이죠?? 코드는 저 버전이 더 깔끔해서 수정했습니다~!

Reply:
Promise.all 을 걸어도 의미가 없다는 뜻이었습니다 !!

```
    await Promise.all(
      userIds.map(async (userId) =>
        await io.in(`user-${userId}`).emit("chat_update", {
          roomId,
        })
      )
    );
```
`io.in`이 비동기를 지원하는지는 모르겠지만 위 코드처럼 수정되어야 `Promise.all`을 사용하는 의미가 있겠네요 ㅎ

Reply:
@chlehdwon

Reply:
바보였군요 제가 이해했어요~!